log/log.go

```diff
@@ -35,36 +43,36 @@ func Warnf(format string, args ...interface{}) {
 	if level < WarnLevel {
 		return
 	}
-	out.WriteString(color.YellowString("WARN "))
-	out.WriteString(fmt.Sprintf(format, args...))
-	out.WriteString("\n")
+	writeOrPanic(color.YellowString("WARN "))
+	writeOrPanic(fmt.Sprintf(format, args...))
+	writeOrPanic("\n")
```

Comment:
All the write calls in this file are for logs. Logs end up being a bit of a special case because if a log fails there ends up not really being any other good way to report the problem. The panic output is going to end up being sent to the same file handle as the logs, so if we can't write logs, then the user also won't see the panic output.

That leaves the question of: "if we fail to log, should the program continue to run or should it exit".  I guess there is an argument to be made either way. I think I'm still slightly on the side of "let the program continue until it hits an error that prevents it from performing its primary function.

Reply:
I'm from the "if something goes wrong, leave a smoking crater" school of thought.

Article by me: [Smoking craters are good](https://docwhat.org/smoking-craters-are-good)